File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/comment/uncomment/direct/Java/C++/s967693306_commented.cpp
#include <iostream>
#include <unordered_map>
#include <string>

using namespace std;

const long long INF = LLONG_MAX / 2;
const int MOD = 1'000'000'007;
const int SIZE = 1'000'000;

long long fac[SIZE];
long long inv[SIZE];
long long finv[SIZE];

class FastScanner {
private:
    istream& in;
    char buffer[1024];
    int ptr = 0;
    int buflen = 0;

    bool hasNextByte() {
        if (ptr < buflen) return true;
        ptr = 0;
        in.read(buffer, sizeof(buffer));
        buflen = in.gcount();
        return buflen > 0;
    }

    int readByte() {
        if (hasNextByte()) return buffer[ptr++];
        return -1;
    }

    static bool isPrintableChar(int c) {
        return 33 <= c && c <= 126;
    }

public:
    FastScanner(istream& input = cin) : in(input) {}

    bool hasNext() {
        while (hasNextByte() && !isPrintableChar(buffer[ptr])) ptr++;
        return hasNextByte();
    }

    string next() {
        if (!hasNext()) throw runtime_error("No more tokens");
        string sb;
        int b = readByte();
        while (isPrintableChar(b)) {
            sb.push_back(static_cast<char>(b));
            b = readByte();
        }
        return sb;
    }

    long long nextLong() {
        if (!hasNext()) throw runtime_error("No more tokens");
        long long n = 0;
        bool minus = false;
        int b = readByte();
        if (b == '-') {
            minus = true;
            b = readByte();
        }
        if (b < '0' || '9' < b) throw runtime_error("Invalid number format");
        while (true) {
            if ('0' <= b && b <= '9') {
                n *= 10;
                n += b - '0';
            } else if (b == -1 || !isPrintableChar(b)) {
                return minus ? -n : n;
            } else {
                throw runtime_error("Invalid number format");
            }
            b = readByte();
        }
    }

    int nextInt() {
        long long nl = nextLong();
        if (nl < INT_MIN || nl > INT_MAX) throw runtime_error("Number out of range");
        return static_cast<int>(nl);
    }

    double nextDouble() {
        return stod(next());
    }
};

class Main {
public:
    void solve() {
        FastScanner sc;
        int n = sc.nextInt();
        int m = sc.nextInt();
        unordered_map<int, int> map;

        for (int i = 0; i < m; i++) {
            int s = sc.nextInt();
            int c = sc.nextInt();
            if (map.find(s) != map.end() && map[s] != c) {
                cout << -1 << endl;
                exit(0);
            }
            map[s] = c;
        }

        for (int i = 0; i < 1000; i++) {
            string s = to_string(i);
            bool flag = true;
            if (s.length() < n) continue;
            for (int j = 0; j < n; j++) {
                if ((map.find(j + 1) != map.end() && map[j + 1] != s[j] - '0') || (map.find(j + 1) == map.end() && s[j] - '0' != 0)) {
                    if (map.find(j + 1) == map.end() && j == 0 && s[j] - '0' == 1) {
                        continue;
                    }
                    flag = false;
                    break;
                }
            }
            if (flag && s.length() == n) {
                cout << s << endl;
                exit(0);
            }
        }

        cout << -1 << endl;
    }

    long long gcd(long long a, long long b) { // return aとbの最大公約数
        if (b == 0) return a;
        return gcd(b, a % b);
    }

    long long lcm(long long a, long long b) { // return aとbの最小公倍数
        return a * b / gcd(a, b);
    }

    long long inv(long long a) { // return aの逆元 (mod MOD)
        return pow(a, MOD - 2);
    }

    long long pow(long long a, long long r) { // return a^r (mod MOD)
        long long sum = 1;
        while (r > 0) {
            if ((r & 1) == 1) { // 2進数表記で末尾1の時
                sum *= a;
                sum %= MOD;
            }
            a *= a;
            a %= MOD;
            r >>= 1;
        }
        return sum;
    }

    long long modFact(long long n) { // retur n! (mod MOD)
        if (n == 0) return 1;
        return n * modFact(n - 1) % MOD;
    }

    long long fact(long long n) { // return n!
        if (n == 0) return 1;
        return n * fact(n - 1);
    }

    void initCOMB() {
        fac[0] = fac[1] = 1;
        inv[1] = 1;
        finv[0] = finv[1] = 1;
        for (int i = 2; i < SIZE; i++) {
            fac[i] = fac[i - 1] * i % MOD;
            inv[i] = MOD - inv[MOD % i] * (MOD / i) % MOD;
            finv[i] = finv[i - 1] * inv[i] % MOD;
        }
    }

    long long modComb(int n, int r) { // return nCr (先にinitCOMB()必要)
        if (n < r || n < 0 || r < 0) return 0;
        return fac[n] * finv[r] % MOD * finv[n - r] % MOD;
    }

    long long comb(long long n, long long r) { // return nCr
        long long num = 1;
        for (long long i = 1; i <= r; i++) {
            num = num * (n - i + 1) / i;
        }
        return num;
    }

    bool isPrime(long long a) { // aの素数判定
        if (a <= 1) return false;
        for (int i = 2; i * i <= a; i++) {
            if (a % i == 0) return false;
        }
        return true;
    }

    string nextPermutation(string s) { // return sの次の順列
        vector<char> list;
        for (int i = 0; i < s.length(); i++) list.push_back(s[i]);

        int pivotPos = -1;
        char pivot = 0;
        for (int i = list.size() - 2; i >= 0; i--) {
            if (list[i] < list[i + 1]) {
                pivotPos = i;
                pivot = list[i];
                break;
            }
        }

        if (pivotPos == -1 && pivot == 0) return "";

        int L = pivotPos + 1;
        int R = list.size() - 1;
        int minPos = -1;
        char min = CHAR_MAX;
        for (int i = R; i >= L; i--) {
            if (pivot < list[i]) {
                if (list[i] < min) {
                    min = list[i];
                    minPos = i;
                }
            }
        }

        swap(list[pivotPos], list[minPos]);
        sort(list.begin() + L, list.begin() + R + 1);

        string result(list.begin(), list.end());
        return result;
    }

    bool nextPermutation(vector<long long>& a) {
        for (int i = a.size() - 1; i > 0; i--) {
            if (a[i - 1] < a[i]) {
                int swapIndex = find(a[i - 1], a, i, a.size() - 1);
                swap(a[swapIndex], a[i - 1]);
                sort(a.begin() + i, a.end());
                return true;
            }
        }
        return false;
    }

    int find(long long dest, const vector<long long>& a, int s, int e) {
        if (s == e) return s;
        int m = (s + e + 1) / 2;
        return a[m] <= dest ? find(dest, a, s, m - 1) : find(dest, a, m, e);
    }

    void elimination(vector<vector<int>>& a, vector<int>& b) {
        int n = a.size();
        double f;
        for (int k = 0; k < n - 1; k++) {
            for (int i = k + 1; i < n; i++) {
                f = -a[i][k] / static_cast<double>(a[k][k]);
                for (int j = k + 1; j < n; j++) {
                    a[i][j] += f * a[k][j];
                }
                b[i] += f * b[k];
            }
            for (int i = n - 1; i >= 0; i--) {
                for (int j = i + 1; j < n; j++) {
                    b[i] -= a[i][j] * b[j];
                }
                b[i] /= a[i][i];
            }
        }
    }
};

int main() {
    Main().solve();
    return 0;
} // End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/multiline/uncomment/direct/Java/Python/s967693306_commented.py
import sys

class FastScanner:
    def __init__(self):
        self.tokens = []
        self.n = 0

    def next(self):
        if self.n < len(self.tokens):
            self.n += 1
            return self.tokens[self.n-1]
        else:
            self.tokens = input().split()
            self.n = 1
            return self.tokens[0]

    def next_int(self):
        return int(self.next())

    def next_long(self):
        return int(self.next())

    def next_double(self):
        return float(self.next())


class Main:
    INF = sys.maxsize // 2
    MOD = 1_000_000_007
    SIZE = 1_000_000

    def __init__(self):
        self.fac = [0] * self.SIZE
        self.inv = [0] * self.SIZE
        self.finv = [0] * self.SIZE
        self.sc = FastScanner()

    def solve(self):
        n = self.sc.next_int()
        m = self.sc.next_int()
        map_dict = {}

        for _ in range(m):
            s = self.sc.next_int()
            c = self.sc.next_int()
            if s in map_dict and map_dict[s] != c:
                print(-1)
                sys.exit(0)
            map_dict[s] = c

        for i in range(1000):
            s = str(i)
            flag = True
            if len(s) < n:
                continue
            for j in range(n):
                if (j + 1 in map_dict and map_dict[j + 1] != int(s[j])) or (j + 1 not in map_dict and int(s[j]) != 0):
                    if j == 0 and j + 1 not in map_dict and int(s[j]) == 1:
                        continue
                    flag = False
                    break
            if flag and len(s) == n:
                print(s)
                sys.exit(0)

        print(-1)

    def gcd(self, a, b):
        return a if b == 0 else self.gcd(b, a % b)

    def lcm(self, a, b):
        return a * b // self.gcd(a, b)

    def inv(self, a):
        return pow(a, self.MOD - 2, self.MOD)

    def pow(self, a, r):
        sum_val = 1
        while r > 0:
            if r & 1:
                sum_val = sum_val * a % self.MOD
            a = a * a % self.MOD
            r >>= 1
        return sum_val

    def mod_fact(self, n):
        return 1 if n == 0 else n * self.mod_fact(n - 1) % self.MOD

    def fact(self, n):
        return 1 if n == 0 else n * self.fact(n - 1)

    def init_comb(self):
        self.fac[0] = self.fac[1] = 1
        self.inv[1] = 1
        self.finv[0] = self.finv[1] = 1
        for i in range(2, self.SIZE):
            self.fac[i] = self.fac[i - 1] * i % self.MOD
            self.inv[i] = self.MOD - self.inv[self.MOD % i] * (self.MOD // i) % self.MOD
            self.finv[i] = self.finv[i - 1] * self.inv[i] % self.MOD

    def mod_comb(self, n, r):
        if n < r or n < 0 or r < 0:
            return 0
        return self.fac[n] * self.finv[r] % self.MOD * self.finv[n - r] % self.MOD

    def comb(self, n, r):
        num = 1
        for i in range(1, r + 1):
            num = num * (n - i + 1) // i
        return num

    def is_prime(self, a):
        if a <= 1:
            return False
        for i in range(2, int(a ** 0.5) + 1):
            if a % i ==  0:
                return False
        return True

    def next_permutation(self, s):
        list_s = list(s)
        pivot_pos = -1
        pivot = 0
        for i in range(len(list_s) - 2, -1, -1):
            if list_s[i] < list_s[i + 1]:
                pivot_pos = i
                pivot = list_s[i]
                break

        if pivot_pos == -1 and pivot == 0:
            return None

        l = pivot_pos + 1
        r = len(list_s) - 1
        min_pos = -1
        min_char = chr(ord('z') + 1)
        for i in range(r, l - 1, -1):
            if pivot < list_s[i]:
                if list_s[i] < min_char:
                    min_char = list_s[i]
                    min_pos = i

        list_s[pivot_pos], list_s[min_pos] = list_s[min_pos], list_s[pivot_pos]
        list_s[l:r + 1] = sorted(list_s[l:r + 1])

        return ''.join(list_s)

    def next_permutation_array(self, a):
        for i in range(len(a) - 1, 0, -1):
            if a[i - 1] < a[i]:
                swap_index = self.find(a[i - 1], a, i, len(a) - 1)
                a[swap_index], a[i - 1] = a[i - 1], a[swap_index]
                a[i:] = sorted(a[i:])
                return True
        return False

    def find(self, dest, a, s, e):
        if s == e:
            return s
        m = (s + e + 1) // 2
        return self.find(dest, a, s, m - 1) if a[m] <= dest else self.find(dest, a, m, e)

    def elimination(self, a, b):
        n = len(a)
        for k in range(n - 1):
            for i in range(k + 1, n):
                f = -a[i][k] / a[k][k]
                for j in range(k + 1, n):
                    a[i][j] += f * a[k][j]
                b[i] += f * b[k]
            for i in range(n - 1, -1, -1):
                for j in range(i + 1, n):
                    b[i] -= a[i][j] * b[j]
                b[i] /= a[i][i]


if __name__ == "__main__":
    main = Main()
    main.solve()  # End of Code
